Sources/TensorFlow/Operators/Math.swift

```diff
@@ -2748,6 +2748,9 @@ public func matmul<Scalar: Numeric>(
     _ rhs: Tensor<Scalar>,
     transposed transposeRhs: Bool = false
 ) -> Tensor<Scalar> {
+    precondition(lhs.rank >= 2, "The input must have rank greater than or equal to 2.")
+    precondition(rhs.rank >= 2, "The input must have rank greater than or equal to 2.")
+    precondition(lhs.shape[1] == rhs.shape[0] , "Dimensions must be equal,but are \(lhs.shape[1]) and \(rhs.shape[0]).")
```

Comment:
Please keep lines under 100 columns:

```suggestion
    precondition(
      lhs.shape[1] == rhs.shape[0],
      """
      Dimension mismatch: lhs.shape[1] is not equal to rhs.shape[0] \
      (\(lhs.shape) vs \(rhs.shape)).
      """
    )
```